sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/SynchronizedAccessor.java

```diff
@@ -38,12 +38,11 @@ public SynchronizedAccessor(Supplier<T> supplier) {
                 return Mono.just(cache);
             }
             if (!wip.getAndSet(true)) {
-                try {
-                    cache = supplier.get();
-                    sink.next(cache);
-                } catch (Exception e) {
-                    sink.error(e);
-                }
+                return supplier.get()
+                    .doOnNext(val -> cache = val)
+                    .doOnNext(sink::next)
+                    .doOnError(sink::error)
+                    .then(Mono.defer(() -> replayProcessor.next()));
```

Comment:
This whole class can be simplified to just this:

```java
public class SynchronizedAccessor<T> {
    private final Mono<T> monoCache;

    public SynchronizedAccessor(Supplier<Mono<T>> supplier) {
        monoCache = supplier.get().cache();
    }

    /**
     * Get the value from the configured supplier.
     *
     * @return the output {@code T}
     */
    public Mono<T> getValue() {
        return monoCache;
    }
}
```

Reply:
That wouldn't work in this scenario.
We want only the first thread to enter and extract the value and cache and other threads to do a non-blocking wait and use the cached value by the first thread.

Reply:
Ideally, supplier should not be invoked at constructor level, but assuming the Mono is wrapped in Mono.defer() this will work fine.
updated.